### ISSUE TITLE
[0.9.x] cannot start pimatic

### ISSUE THREAD

**Issue by andig** [pimatic/pimatic#896]
Coming back to pimatic today it fails to start:

	pi@raspberrypi:~/pimatic $ sudo node pimatic.js
	module.js:327
	    throw err;
	    ^

	Error: Cannot find module 'fs.extra'
	    at Function.Module._resolveFilename (module.js:325:15)
	    at Function.Module._load (module.js:276:25)
	    at Module.require (module.js:353:17)
	    at require (internal/module.js:12:17)
	    at Object.<anonymous> (/home/pi/pimatic/node_modules/pimatic/coffee-cache.js:27:14)
	    at Module._compile (module.js:409:26)
	    at Object.Module._extensions..js (module.js:416:10)
	    at Module.load (module.js:343:32)
	    at Function.Module._load (module.js:300:12)
	    at Module.require (module.js:353:17)

The line in question seems to be in coffee-cache.js:

    var fs     = require('fs.extra');

`npm update` doesn't change anything. Might this be a missing dependency somewhere?

**Comment by andig** [comment]
Mhm. `npm info` is entirely unhappy:


	npm ERR! missing: byline@~4.2.1, required by pimatic@0.8.107
	npm ERR! missing: jsonlint@git+https://github.com/sweetpi/jsonlint.git#master, required by pimatic@0.8.107
	npm ERR! missing: connect-timeout@^1.6.2, required by pimatic@0.8.107
	npm ERR! missing: cross-spawn@^0.2.3, required by pimatic@0.8.107
	npm ERR! missing: daemon@1.1.0, required by pimatic@0.8.107
	npm ERR! missing: date-format-lite@0.5.0, required by pimatic@0.8.107
	npm ERR! missing: decl-api@0.0.20, required by pimatic@0.8.107
	npm ERR! missing: express@3.21.2, required by pimatic@0.8.107
	npm ERR! missing: fs.extra@1.2.1, required by pimatic@0.8.107
	npm ERR! missing: human-format@^0.2.0, required by pimatic@0.8.107
	npm ERR! missing: i18n@0.4.*, required by pimatic@0.8.107
	npm ERR! missing: jayschema@~0.2.7, required by pimatic@0.8.107
	npm ERR! missing: knex@0.9.0, required by pimatic@0.8.107
	npm ERR! missing: lodash@2.4.1, required by pimatic@0.8.107
	npm ERR! missing: logrotate-stream@^0.2.3, required by pimatic@0.8.107
	npm ERR! missing: moment@~2.6.0, required by pimatic@0.8.107
	npm ERR! missing: relaxed-json@~0.2.4, required by pimatic@0.8.107
	npm ERR! missing: request-promise@^0.4.0, required by pimatic@0.8.107
	npm ERR! missing: semver@2.2.1, required by pimatic@0.8.107

Seems that installing `pimatic-fritz` has brought in the old pimatic core somehow?

**Comment by andig** [comment]
Confired working. 

**tl;dr**

Module authors make sure peer dependency doesn't force pimatic 0.8. `^0.8.x` is not sufficient imho it needs be `^0.8.x||^0.9.x`.

**Comment by mwittig** [comment]
> Module authors make sure peer dependency doesn't force pimatic 0.8. ^0.8.x is not sufficient imho it needs be ^0.8.x||^0.9.x.

It is sufficient to defined the peer dependency as follows:
``` 
"pimatic": ">0.8.91 <1.0.0"
```

**Comment by andig** [comment]
Thats about equivalent I guess. Fine to close unless that needs to go for docs somewhere?

**Comment by mwittig** [comment]
Has been documented as part of the "Checklist for developers" - https://github.com/pimatic/pimatic/issues/859
